CONTRIBUTING.md

```diff
@@ -1,61 +1,520 @@
+  * [Amplify History and Ethos](#amplify-history-and-ethos)
+    + [Amplify Design](#amplify-design)
+  * [Getting Started](#getting-started)
+  * [Tools](#tools)
+  * [Workflows](#workflows)
+    + [Adding code to support a new feature](#adding-code-to-support-a-new-feature)
+  * [Reporting Bugs/Feature Requests](#reporting-bugs-feature-requests)
+  * [Contributing via Pull Requests](#contributing-via-pull-requests)
 
-## Reporting Bugs/Feature Requests
 
-We welcome you to use the GitHub issue tracker to report bugs or suggest features.
+## Amplify History and Ethos
+AWS Amplify aims to simplify the experience of developing cloud
+integrations on Android. Amplify codifies best practices through
+programmatic interfaces that reduce the effort needed to interact with
+cloud resources.
+
+First and foremost, Amplify exposes to you _what_ things do, and then
+_how_ best to do them. The "_what_" is a functional use case, and
+"_how_" is an opinionated implementation that you can override with
+"escape hatches." This will allow you to have higher velocity and build
+better applications by focusing less on implementation choices.
+
+Secondly, Amplify is a manifestation of
+[The Rule of Least Power](https://en.wikipedia.org/wiki/Rule_of_least_power)
+when developing against AWS. This means it encourages architectural and
+programmatic best practices that enable you to start quickly.  This
+shows by encouraging certain services (API Gateway usage vs. direct
+DynamoDB interaction) or certain connection patterns (Circuit breaker,
+retry counts and throttle up/down).
+
+__Opinionated implementations__: There are many ways to interface with AWS
+Services. Certain service interactions are favored over others. For instance, if
+sending and receiving JSON, we would prefer an API Gateway endpoint to other
+mechanisms. Amplify will programmatically help optimize for cost and
+performance through library decisions.
+
+__Declarative actions__: Amplify will provide you a reference to a generic client
+object and offer an ability to perform common actions: "RegisterUser", "Login",
+"SendObject", "UpdateObject", "StreamData". By default, you should not need to
+worry about AWS-service-specific API operations like `putItem()` with a unique
+hash -- or even which HTTP verbs to use.
+
+__Cascading service interactions__: Certain actions in a declarative style can have
+overlapping or ambiguous AWS Service implementations. With an opinionated
+implementation we can decide which Services are "primary" and which are
+"secondary" depending on what is configured. For instance sending an image will
+prefer S3 over API Gateway..
+
+__Simple, standard data objects__: Sending & Receiving data to AWS Services can
+have many parameters, which tend to show up in the SDKs. These are abstracted
+and inferred, where possible, with simple JSON that the implementation can
+reason about. Standard parameters (bucket names, stream names, partition keys,
```

Comment:
I'd remove the last part ",with simple JSON..." as it was an older JS-specific statement